### PR TITLE
fix: if preprocessor is async function and doesn't return a content t…

### PR DESCRIPTION
**Pull request by anthony-redFox**
…hen await donePromise

### CODE REVIEW

**Comment by anthony-redFox** [comment]
@johnjbarton please take a look

File: lib/preprocessor.js [hunk 1]
```diff
@@ -24,7 +24,13 @@ function executeProcessor (process, file, content) {
       }
     }
   })
-  return process(content, file, done) || donePromise
+
+  return (process(content, file, done) || Promise.resolve()).then((content) => {
+    if (content) {
```
Comment (johnjbarton) [inline lib/preprocessor.js]: I found this code confusing to read.  Let's add a comment here:
`// async process correctly returned content/`

File: lib/preprocessor.js [hunk 2]
```diff
@@ -24,7 +24,13 @@ function executeProcessor (process, file, content) {
...
+      return content
+    }
```
Comment (johnjbarton) [inline lib/preprocessor.js]: `// process called done() (Either old sync api or an async function that did not return content)`

File: test/unit/preprocessor.spec.js [hunk 3]
```diff
@@ -80,6 +80,27 @@ describe('preprocessor', () => {
     })
   })
 
+  it('should get content if preprocessor is async function', (done) => {
```
Comment (johnjbarton) [inline test/unit/preprocessor.spec.js]: ...is an async function still calling done()'

**Comment by anthony-redFox** [comment]
@johnjbarton  Thanks for review. Add new unit tests and comments

**Pull request merged**